schemachanger: implement ALTER TABLE .. RENAME in declarative schemachanger

rafiss opened the pull request:
See individual commits to make this easier to review.

This preserves all the validations and error handling from the legacy schema
changer.

###  scexec: add deferred mutation to rename TTL schedule

This will make sure the TTL schedule name matches the new table name
when a table is renamed, just as the legacy schema changer does.

### scbuild: handle trigger/procedure dependencies for table renames 

###  lease_test: fix tests that rely on legacy testing knobs

Since RENAME now works in the declarative schema changer, we need to
force these tests to use the legacy schema changer to test the desired
behavior.

fixes #148339 
Release note: None

rafiss committed:
schemachanger: implement ALTER TABLE .. RENAME in declarative schema changer

This preserves all the validations and error handling from the legacy schema
changer.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/152895)
<!-- Reviewable:end -->

rafiss committed:
scexec: add deferred mutation to rename TTL schedule

This will make sure the TTL schedule name matches the new table name
when a table is renamed, just as the legacy schema changer does.

Release note: None

rafiss committed:
scbuild: handle trigger/procedure dependencies for table renames

Release note: None

rafiss committed:
lease_test: fix tests that rely on legacy testing knobs

Sicne RENAME now works in the declarative schema changer, we need to
force these tests to use the legacy schema changer to test the desired
behavior.

Release note: None

fqazi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

Should we include a TTL schedule in the end to end test?

> @fqazi reviewed 1 of 12 files at r1, 4 of 13 files at r2, 5 of 25 files at r8, 3 of 23 files at r10, 50 of 50 files at r13, 49 of 49 files at r14, 13 of 13 files at r15, 7 of 7 files at r16, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/152895#-:-O_WtHj2--bWUDQlAbRr:b-tbb0ja)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @aerfrei and @jeffswenson)



<!-- Sent from Reviewable.io -->

rafiss commented:
TFTR! I added that to the test

bors r+
